x-pack/plugins/enterprise_search/public/applications/app_search/components/search_ui/components/search_ui_form.tsx

```diff
@@ -0,0 +1,205 @@
+);
+const FILTER_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldLabel',
+  { defaultMessage: 'Filter fields (Optional)' }
+);
+const FILTER_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldHelpText',
+  { defaultMessage: 'Faceted values rendered as filters and available as query refinement' }
+);
+const FILTER_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.filterFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const SORT_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldLabel',
+  { defaultMessage: 'Sort fields (Optional)' }
+);
+const SORT_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortHelpText',
+  { defaultMessage: 'Used to display result sorting options, ascending and descending' }
+);
+const SORT_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.sortFieldPlaceholder',
+  { defaultMessage: 'Click to select' }
+);
+const URL_FIELD_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldLabel',
+  { defaultMessage: 'URL field (Optional)' }
+);
+const URL_FIELD_HELP_TEXT = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldHelpText',
+  { defaultMessage: "Used as a result's link target, if applicable" }
+);
+const URL_FIELD_PLACEHOLDER = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.urlFieldPlaceholder',
+  { defaultMessage: 'Select a URL field' }
+);
+const GENERATE_PREVIEW_BUTTON_LABEL = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.searchUI.generatePreviewButtonLabel',
+  { defaultMessage: 'Generate a Preview' }
+);
+
+interface Option<T> extends Omit<EuiComboBoxOptionOption<T>, 'value'> {
+  value: string;
+}
+
+export const SearchUIForm: React.FC = () => {
+  const {
+    validFields,
+    validSortFields,
+    validFacetFields,
+    titleField,
+    urlField,
+    facetFields,
+    sortFields,
+  } = useValues(SearchUILogic);
+  const { engineName } = useValues(EngineLogic);
+  const {
+    onActiveFieldChange,
+    onFacetFieldsChange,
+    onSortFieldsChange,
+    onTitleFieldChange,
+    onUrlFieldChange,
+  } = useActions(SearchUILogic);
+
+  const previewHref = generatePreviwUrl(
+    {
+      fromKibana: 'true',
+      titleField,
+      urlField,
+      facets: facetFields,
+      sortFields,
+    },
+    engineName
+  );
+
+  const formatSelectOption = (fieldName: string) => {
+    return { text: fieldName, value: fieldName };
+  };
+  const formatMultiOptions = (fieldNames: string[]) =>
+    fieldNames.map((fieldName) => ({ label: fieldName, text: fieldName, value: fieldName }));
+  const formatMultiOptionsWithEmptyOption = (fieldNames: string[]) => [
+    { label: '', text: '', value: '' },
+    ...formatMultiOptions(fieldNames),
+  ];
```

Comment:
[Not a change request! just me thinking out loud] I wonder if we should DRY out a general util helper for EUI selects/comboboxes where we want the text/value/label to be all the same thing. I know we do this in a few other places (meta engine creation for example?) - I might jot this down as one of my tech debt wishlist items